transactions/demo-rust/build.nix

```diff
@@ -0,0 +1,80 @@
+{ inputs, ... }: {
+  perSystem = { config, inputs', system, pkgs, ... }:
+
+    let
+      rustFlake = inputs.flake-lang.lib."${system}".rustFlake
+        {
+          src = ./.;
+          inherit (inputs) crane;
+          crateName = "demo";
+          nativeBuildInputs =
+            (pkgs.lib.optionals pkgs.stdenv.isLinux [
+              pkgs.pkg-config
+            ]) ++
+            (pkgs.lib.optionals pkgs.stdenv.isDarwin
+              [
+                pkgs.gcc
+                pkgs.darwin.apple_sdk.frameworks.Security
+                pkgs.darwin.apple_sdk.frameworks.SystemConfiguration
+              ]);
+          buildInputs = [
+            pkgs.openssl.dev
+          ];
```

Comment:
Wait, why is this here? Shouldn't this be upstreamed? I don't want to know about this if i want to have my rust project build with darwin. Or do I?